src/System.Private.DataContractSerialization/src/System/Xml/XmlDictionaryString.cs

```diff
@@ -26,7 +26,7 @@ public class XmlDictionaryString
         public XmlDictionaryString(IXmlDictionary dictionary, string value, int key)
         {
             if (dictionary == null)
-                _dictionary = new EmptyStringDictionary();
+                throw System.Runtime.Serialization.DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentNullException("dictionary"));
```

Comment:
This seems like a breaking change although according to MSDN we should be throwing ArgumentNullException. I can also see that we do exactly the same thing (new behavior) on desktop:
http://referencesource.microsoft.com/#System.Runtime.Serialization/System/Xml/XmlDictionaryString.cs,6fc6fdc6bb5abc25

Was it some fairly new fix which we forgot to port to corefx?